docling-server/src/utils.py

```
@@ -0,0 +1,675 @@
+    """Analyze text quality per page and return OCR candidates.
+    
+    ALL pages are OCR candidates - native text layer is not trusted.
+    VLM OCR is used for every page.
+    
+    Returns dict with page_no -> {native_chars, decision, ocr_candidate}
+    """
+    page_text, page_chars = collect_page_text(doc)
+    qualities: Dict[int, Dict[str, any]] = {}
+    pages = getattr(doc, "pages", {})
+
+    print(f"Page quality: ALL pages sent to VLM OCR (native layer bypassed)")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)